R/canned-epipred.R

```diff
@@ -31,20 +31,15 @@ arx_lags_validator <- function(predictors, lags) {
       "You have requested {p} predictor(s) but {l} different lags.",
       i = "Lags must be a vector or a list with length == number of predictors."
     ))
-  }else{
-    if(length(lags) == sum(names(lags) != "", na.rm = TRUE)){
-      if(!all(names(lags) %in% predictors)){
+  } else {
+    if(length(lags) == sum(names(lags) != "", na.rm = TRUE)) {
+      if (all(names(lags) %in% predictors)) {
+        lags <- lags[order(match(names(lags), predictors))]
+      } else {
         cli::cli_abort(
           "The names for the list of lags do not all correspond to the predictors."
```

Comment:
Sorry, a few more corrections:
1. On 35: `if(length(lags)...` should be `if (length(lags)...` with a space between `if` and `(`.
2. For the error, let's say "If `lags` is a named list, then all `predictors` must be present". It would be good to also list which ones are missing. An example of how to do this is `validate_meta_match()` in `epi_check_training_set.R`. (Though there, I see that on line 43 it has "`geo_type`" when it should have `{meta}`)
3. Finally, I think the check on 36 is actually not quite right. What we need is all predictors in names(lags) rather than the other way around.